sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -139,15 +146,25 @@ class OrcFileFormat
       }
     }
 
+    val resultSchema = StructType(requiredSchema.fields ++ partitionSchema.fields)
+    val enableVectorizedReader = sparkSession.sessionState.conf.orcVectorizedReaderEnabled &&
+      supportBatch(sparkSession, resultSchema)
```

Comment:
Whether enabling vectorized reader is not the same as supporting batch. You can enable vectorized reader but not support batch like `ParquetFileFormat` does.

Reply:
Ur, it seems to make the implementation complicated in order to provide slower version.
If you don't mind, I'll proceed that in a next follow-up PR.

Reply:
`supportBatch` already have the condition `conf.orcVectorizedReaderEnabled`, I think here we can just write
```
val enableVectorizedReader = supportBatch(sparkSession, resultSchema)
```

Reply:
Right. It's fixed.